writeaheadlog.go

```diff
@@ -174,9 +218,31 @@ func (w *WAL) recoverWal(data []byte) ([]Update, error) {
 	var firstPages ByTxnNumber
 
 	// Validate metadata
-	if err := readWALMetadata(data[0:]); err != nil {
+	recoveryState, err := readWALMetadata(data[0:])
+
+	if err != nil {
 		return nil, err
 	}
+
+	if recoveryState == recoveryStateClean {
+		return []Update{}, nil
+	}
+
+	// If recoveryState is set to wipe we don't need to recover but we have to
+	// wipe the wal and change the state
+	if recoveryState == recoveryStateWipe {
+		if err := w.wipeWAL(); err != nil {
+			return nil, err
+		}
+		if err := w.logFile.Sync(); err != nil {
+			return nil, err
+		}
+		if err := w.writeRecoveryState(recoveryStateUnclean); err != nil {
+			return nil, err
+		}
+		return []Update{}, w.logFile.Sync()
```

Comment:
You can put all of these steps in to `wipeWAL`, since you need to perform all of these steps any time that you wipe the WAL.